lib/mayaUsd/ufe/UsdTransform3dMatrixOp.cpp

```diff
@@ -128,105 +131,64 @@ class UsdSetMatrix4dUndoableCmd : public UsdUndoableCommand<Ufe::SetMatrix4dUndo
     const Ufe::Matrix4d _newM;
 };
 
-// Helper class to factor out common code for translate, rotate, scale
-// undoable commands.
-class UsdTRSUndoableCmdBase
+// Factor out common code for translate, rotate, scale undoable commands.
+class MatrixOpUndoableCmdBase : public UsdSetXformOpUndoableCommandBase<GfMatrix4d>
 {
-private:
-    UsdGeomXformOp    fOp;
-    const UsdTimeCode fReadTime;
-    const UsdTimeCode fWriteTime;
-    const VtValue     fPrevOpValue;
+    UsdGeomXformOp _op;
 
 public:
-    UsdTRSUndoableCmdBase(
+    MatrixOpUndoableCmdBase(
         const Ufe::Path&      path,
         const UsdGeomXformOp& op,
-        const UsdTimeCode&    writeTime_)
-        : fOp(op)
-        ,
-        // Always read from proxy shape time.
-        fReadTime(getTime(path))
-        , fWriteTime(writeTime_)
-        , fPrevOpValue(getValue(op.GetAttr(), readTime()))
-        , fNewOpValue(fPrevOpValue)
+        const UsdTimeCode&    writeTime)
+        : UsdSetXformOpUndoableCommandBase(path, writeTime)
+        , _op(op)
     {
     }
 
-    void undo() { fOp.GetAttr().Set(fPrevOpValue, fWriteTime); }
-    void redo() { fOp.GetAttr().Set(fNewOpValue, fWriteTime); }
-
-    UsdTimeCode readTime() const { return fReadTime; }
-    UsdTimeCode writeTime() const { return fWriteTime; }
-
-protected:
-    VtValue fNewOpValue;
+    void setValue(const GfMatrix4d& m) override
+    {
+        VtValue v;
+        v = m; // Can assign to VtValue, but can't construct.
+        _op.GetAttr().Set(v, writeTime());
+    }
 };
 
 // Command to set the translation on a scene item by setting a matrix transform
 // op at an arbitrary position in the transform op stack.
-class UsdTranslateUndoableCmd
-    : public Ufe::TranslateUndoableCommand
-    , public UsdTRSUndoableCmdBase
+class MatrixOpTranslateUndoableCmd : public MatrixOpUndoableCmdBase
 {
 public:
-#ifdef UFE_V2_FEATURES_AVAILABLE
```

Comment:
Removed all UFE_V2_FEATURES_AVAILABLE pre-processor conditionals in this file.  This was completely bogus, as UsdTransform3dMatrixOp.cpp is a UFE v2 file.